src/tasks/AppleAppBuilder/Templates/runtime.m

```diff
@@ -201,6 +203,52 @@
 //%DllMap%
 }
 
+int32_t GlobalizationNative_LoadICUData(void * pData);
+
+static int32_t load_icu_data()
+{
+    char path [1024];
+    int res;
+    
+    const char *dname = "icudt.dat";
+    const char *bundle = get_bundle_path ();
+
+    os_log_info (OS_LOG_DEFAULT, "Loading ICU data file '%s'.", dname);
+    res = snprintf (path, sizeof (path) - 1, "%s/%s", bundle, dname);
+    assert (res > 0);
+   
+    char *icu_data = NULL;
+    FILE *fp = fopen(path, "rb");
+    if (fp != NULL) {
+        if (fseek(fp, 0L, SEEK_END) == 0) {
+            long bufsize = ftell(fp);
+
+            if (bufsize == -1) {
+                 os_log_info (OS_LOG_DEFAULT, "Invalid ICU dat file");
+                 exit(1);
+            }
+
+            icu_data = malloc(sizeof(char) * (bufsize + 1));
+
+            if (fseek(fp, 0L, SEEK_SET) != 0) {
+                os_log_info (OS_LOG_DEFAULT, "Unable to seek ICU dat file.");
+                exit(1);
+            }
+
+            size_t len = fread(icu_data, sizeof(char), bufsize, fp);
+            if ( ferror( fp ) != 0 ) {
+                os_log_info (OS_LOG_DEFAULT, "Unable to read ICU dat file");
+                exit(1);
+            }
+        }
+        fclose(fp);
+    }
+
+    free(icu_data);
```

Comment:
aren't you freeing memory you later pass to `GlobalizationNative_LoadICUData` ?
Also can we at least print a warning to the console that "icudt.dat was not found, falling back to invariant mode" in case if doesn't exists?

Reply:
Yep :-).  That's not good.